pkg/vsphere/vsan/host_vsan_internal_system.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vsan
+
+import (
+	"context"
+	"encoding/json"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
+	//	"github.com/vmware/govmomi/vim25/mo"
+	//	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	//	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	DomObject  = "DOM_OBJECT"
+	LsomObject = "LSOM_OBJECT"
+	Policy     = "POLICY"
+	Disk       = "DISK"
+)
+
+type VSANExtAttrs struct {
+	Path string `json:"Object path"`
+}
+
+// TODO: HostVsanInternalSystem should be moved to govmomi
+type HostVsanInternalSystem struct {
+	object.Common
+	cl *vim25.Client
+}
+
+func NewHostVsanInternalSystem(c *vim25.Client, ref types.ManagedObjectReference) *HostVsanInternalSystem {
+	m := HostVsanInternalSystem{
```

Comment:
in this particular case you could just:
```
return &HostVsanInternalSystem{
    Common: object.NewCommon(c, ref),
    cl:     c,
}
```

However, this is a personal preference.